SPEC.md

```diff
@@ -4002,28 +4019,28 @@ Optional whitespace is allowed between the number/expression and the suffix. For
 
 ##### `container`
 
 * Accepted types:
+    * `"*"`: This special value indicates that the runtime engine may use any POSIX-compliant operating environment it wishes to execute the task, whether that be a container or directly in the host environment.
     * `String`: A single container URI.
     * `Array[String]`: An array of container URIs.
+* Default value: `"*"`
 * Alias: `docker`
 
-The `container` key accepts a URI string that describes a location where the execution engine can attempt to retrieve a container image to execute the task.
-
-The user is strongly suggested to specify a `container` for every task. There is no default value for `container`. If `container` is not specified, the execution behavior is determined by the execution engine. Typically, the task is simply executed in the host environment. 
+The `container` attribute accepts a URI string that describes a container resource the execution engine can use when executing the task.
 
-🗑 The ability to omit `container` is deprecated. In WDL 2.0, `container` will be required.
+The user is strongly suggested to specify a `container` for every task. If `container` is not specified, or is specified with the special `"*"` value, the execution behavior is determined by the execution engine. A task that depends on the engine to determine the execution environment should be careful to only use built-in Bash operations and tools specified as mandatory by the [POSIX standard](https://unix.stackexchange.com/questions/228888/what-are-the-posix-mandatory-utilities).
 
-The format of a container URI string is `protocol://location`, where protocol is one of the protocols supported by the execution engine. Execution engines must, at a minimum, support the `docker://` protocol, and if no protocol is specified, it is assumed to be `docker://`. An execution engine should ignore any URI with a protocol it does not support.
+The format of a container URI is `protocol://location`, where `protocol` is one of the protocols supported by the execution engine. Execution engines must, at a minimum, support the `docker` protocol. If only `location` is specified, the protocol is assumed to be `docker`. An execution engine should ignore any URI with a protocol it does not support.
 
-Container source locations should use the syntax defined by the individual container repository. For example an image defined as `ubuntu:latest` would conventionally refer a docker image living on `DockerHub`, while an image defined as `quay.io/bitnami/python` would refer to a `quay.io` repository.
+A container location uses the syntax defined by the container repository. For example, the URI `ubuntu:latest` refers to a Docker image hosted on `DockerHub`, while the URI `quay.io/bitnami/python` refers to an image in a `quay.io` repository.
```

Comment:
Can we state that URI patterns with SHA sums are allowed (perhaps even encouraged when repository tags are mutable)?

Reply:
good suggestion